interface/main/mobile/m_flow.php

```diff
@@ -0,0 +1,677 @@
+                            $appointment[$row['msg_type']]['stage'] = "CANCELLED";
+                            $icon_here[$row['msg_type']] = '';
+                        } elseif ($row['msg_reply'] == "FAILED") {
+                            $appointment[$row['msg_type']]['stage'] = "FAILED";
+                            $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['FAILED']['html'];
+                        } elseif (($row['msg_reply'] == "CONFIRMED") || ($appointment[$row['msg_type']]['stage'] == "CONFIRMED")) {
+                            $appointment[$row['msg_type']]['stage'] = "CONFIRMED";
+                            $icon_here[$row['msg_type']]  = $icons[$row['msg_type']]['CONFIRMED']['html'];
+                        } elseif ($row['msg_type'] == "NOTES") {
+                            $CALLED = "1";
+                            $FINAL = $icons['NOTES']['CALLED']['html'];
+                            $icon_CALL = str_replace("Call Back: COMPLETED", attr(oeFormatShortDate($row['msg_date'])) . " :: " . xla('Callback Performed') . " | " . xla('NOTES') . ": " . $row['msg_extra_text'] . " | ", $FINAL);
+                            continue;
+                        } elseif (($row['msg_reply'] == "READ") || ($appointment[$row['msg_type']]['stage'] == "READ")) {
+                            $appointment[$row['msg_type']]['stage'] = "READ";
+                            $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['READ']['html'];
+                        } elseif (($row['msg_reply'] == "SENT") || ($appointment[$row['msg_type']]['stage'] == "SENT")) {
+                            $appointment[$row['msg_type']]['stage'] = "SENT";
+                            $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['SENT']['html'];
+                        } elseif (($row['msg_reply'] == "To Send") || (empty($appointment['stage']))) {
+                            if (($appointment[$row['msg_type']]['stage'] != "CONFIRMED") &&
+                                ($appointment[$row['msg_type']]['stage'] != "READ") &&
+                                ($appointment[$row['msg_type']]['stage'] != "SENT") &&
+                                ($appointment[$row['msg_type']]['stage'] != "FAILED")) {
+                                $appointment[$row['msg_type']]['stage'] = "QUEUED";
+                                $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['SCHEDULED']['html'];
+                            }
+                        }
+                        //these are additional icons if present
+                        if (($row['msg_reply'] == "CALL") && (!$CALLED)) {
+                            $icon_here = '';
+                            $icon_4_CALL = $icons[$row['msg_type']]['CALL']['html'];
+                            $icon_CALL = "<span onclick=\"doCALLback('" . attr($date_squash) . "','" . attr($appointment['eid']) . "','" . attr($appointment['pc_cattype']) . "')\">" . $icon_4_CALL . "</span>
```

Comment:
We have a more robust way to deal with items that are embedded in the onclick. This is a weird scenaria since it is combining both js and html elements, and looks like so when done (note that quotes are removes since the js escaping does this):
```
$icon_CALL = "<span onclick=\"doCALLback(" . attr_js$date_squash) . "," .attr_js($appointment['eid']) . "," . attr_js($appointment['pc_cattype']) . ")\">" . $icon_4_CALL . "</span>
```